xbmc/cores/RetroPlayer/RetroPlayer.cpp

```diff
@@ -106,18 +113,43 @@ bool CRetroPlayer::OpenFile(const CFileItem& file, const CPlayerOptions& options
 
   if (bSuccess)
   {
-    if (file.m_lStartOffset == STARTOFFSET_RESUME && file.HasGameInfoTag())
+    std::string savestatePath = CSavestateUtils::MakeMetadataPath(file.GetPath());
+
+    CSavestate save;
+    if (save.Deserialize(savestatePath))
     {
-      std::string redactedSavestatePath = CURL::GetRedacted(file.GetGameInfoTag()->GetSavestate());
+      // Check if game client is the same
+      if (save.GameClient() != m_gameClient->ID())
+      {
+        ADDON::AddonPtr addon;
+        if (ADDON::CAddonMgr::GetInstance().GetAddon(save.GameClient(), addon))
+        {
+          // "Opening file"
+          // "The savestate can only be loaded with \"{0:s}\". Erase savestate and continue?"
+          // "Cancel"
+          // "Erase save"
```

Comment:
This isn't correct anymore. I wouldn't write this here anyway because it can get out of sync quickly.